llama-index-integrations/vector_stores/llama-index-vector-stores-baiduvectordb/llama_index/vector_stores/baiduvectordb/base.py

```diff
@@ -0,0 +1,440 @@
+DEFAULT_TABLE_NAME = "llama_default_table"
+DEFAULT_TIMEOUT_IN_MILLS: int = 30*1000
+
+DEFAULT_PARTITION = 1
+DEFAULT_REPLICA = 3
+DEFAULT_INDEX_TYPE = "HNSW"
+DEFAULT_METRIC_TYPE = "L2"
+
+DEFAULT_HNSW_M = 16
+DEFAULT_HNSW_EF_CONSTRUCTION = 200
+DEFAULT_HNSW_EF = 10
+
+FIELD_ID: str = "id"
+FIELD_VECTOR: str = "vector"
+FIELD_METADATA: str = "metadata"
+INDEX_SUFFIX: str = "_index"
+INDEX_VECTOR: str = "vector_index"
+
+VALUE_NONE_ERROR = "Parameter `{}` can not be None."
+VALUE_RANGE_ERROR = "The value of parameter `{}` must be within {}."
+NOT_SUPPORT_INDEX_TYPE_ERROR = (
+    "Unsupported index type: `{}`, supported index types are {}"
+)
+NOT_SUPPORT_METRIC_TYPE_ERROR = (
+    "Unsupported metric type: `{}`, supported metric types are {}"
+)
+
+
+def _try_import() -> None:
+    try:
+        import pymochow
+    except ImportError:
+        raise ImportError(
+            "`pymochow` package not found, please run `pip install pymochow`"
+        )
+
+class TableField:
+    name: str
+    data_type: str = "STRING"
+
+    def __init__(self, name: str, data_type: str = "STRING"):
+        self.name = name
+        self.data_type = "STRING" if data_type is None else data_type
+
+
+class TableParams:
+    """Baidu VectorDB table params.
+    See the following documentation for details:
+    https://cloud.baidu.com/doc/VDB/s/mlrsob0p6
+    
+    Args:
+        dimension int: The dimension of vector.
+        replication int: The number of replicas in the table.
+        partition int: The number of partitions in the table.
+        index_type (Optional[str]): HNSW, FLAT... Default value is "HNSW"
+        metric_type (Optional[str]): L2, COSINE, IP. Default value is "L2"
+        drop_exists (Optional[bool]): Delete the existing Table. Default value is False.
+        vector_params (Optional[Dict]):
+          if HNSW set parameters: `M` and `efConstruction`, for example `{'M': 16, efConstruction: 200}`
+          default is HNSW
+        filter_fields: Optional[List[str]]: Set the fields for filtering, The
+        fields used for filtering must have a value in every row of the table
+        and cannot be null.
+          for example: ['author', 'age']
+          This can be used when calling the query method：
+             store.add([
+                TextNode(..., metadata={'age'=23, 'name'='name1'})
+            ])
+             ...
+             query = VectorStoreQuery(...)
+             store.query(query, filter="age > 20 and age < 40 and name = 'name1'")
+    """
+
+    def __init__(
+        self,
+        dimension: int,
+        table_name: str = DEFAULT_TABLE_NAME,
+        replication: int = DEFAULT_REPLICA,
+        partition: int = DEFAULT_PARTITION,
+        index_type: str = DEFAULT_INDEX_TYPE,
+        metric_type: str = DEFAULT_METRIC_TYPE,
+        drop_exists: Optional[bool] = False,
+        vector_params: Optional[Dict] = None,
+        filter_fields: Optional[List[TableField]] = [],
```

Comment:
`[]` as a default can produce unexpected behaviors. https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument
```suggestion
        filter_fields: Optional[List[TableField]] = None,
    ):
        if filter_fields is None:
            filter_fields = []
```

Reply:
I greatly appreciate your prompt response. The issue has been resolved on my end. Would you be so kind as to check if there are any further comments or concerns? Thank you very much. @skvrd